content/blog/2016/2016-10-27-xunit-reporting.adoc

```diff
@@ -0,0 +1,225 @@
+---
+layout: post
+title: "xUnit and Pipeline"
+tags:
+- pipeline
+- plugins
+- xunit
+- nightwatch
+author: lnewman
+---
+
+NOTE: This is a guest post by link:https://github.com/bitwiseman[Liam Newman],
+Technical Evangelist at link:http://cloudbees.com[CloudBees].
+
+The
+link:https://wiki.jenkins-ci.org/display/JENKINS/JUnit+Plugin[JUnit plugin]
+is the go-to test result reporter for many Jenkins projects,
+but the it is not the only one available.  The
+link:https://wiki.jenkins-ci.org/display/JENKINS/xUnit+Plugin[xUnit plugin]
+is a viable alternative that supports JUnit and many other test result file formats.
+
+=== Introduction
+
+No matter the project, you need to gather and report test results.
+JUnit is one of the most widely supported formats for recording test results.
+For a scenarios where your tests are stable and your framework can produce JUnit output,
+this makes the JUnit plugin ideal for reporting results in Jenkins.
+It will consume results from a specified file or path, create a report,
+and if it finds test failures it will set the the job state to "unstable" or "failed".
+
+There are also plenty of scenarios where the JUnit plugin is not enough.
+If your project has some failing tests that will take some time to fix,
+or if there are some flaky tests,
+the JUnit plugin's simplistic view of test failures may be difficult to work with.
+
+No problem, the Jenkins plugin model lets us replace the JUnit
+plugin functionality with similar
+functionality from another plugin and Jenkins Pipeline lets us do this in safe
+stepwise fashion where we can test and debug each of our changes.
+
+In this article, I will show you how to replace the JUnit plugin with the
+xUnit plugin in Pipeline code to address a few common test reporting scenarios.
+
+=== Initial Setup
+
+I'm going to use the Nightwatch.js sample project from my
+link:/blog/2016/08/29/sauce-pipeline/[previous post] to demonstrate a couple
+common scenarios that the xUnit handles better.
+I already have the latest JUnit and xUnit plugins installed on my Jenkins server.
+
+I'll be keeping my changes in my fork of the Nightwatch.js sample project on GitHub, under the "blog/xunit" branch.
+
+Here's what the Jenkinsfile looked like at the end of that previous post and what
+the report page looks like after a few runs:
+
+[source,groovy]
+----
+// TODO: Jenkinsfile
+
+----
+
+TODO: Image showing results reporting.
+
+=== Switching from JUnit to xUnit
+
+I'll start by replacing JUnit with xUnit in my pipeline.
+I use the Snippet Generator to create the step with the right parameters.
+The main downside of using the xUnit plugin is that while it is Pipeline compatible,
```

Comment:
If you have some time, perhaps you could file an issue for the xunit plugin maintainer and show them the `@Symbol` annotation ([documentation here](https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md#defining-symbols))

It should be really easy to add an `xunit` symbol. `junit` afterall is just a `@Symbol` annotation :D